app/src/androidTest/java/ch/epfl/sweng/swenggolf/ProfileActivityTest.java

```diff
@@ -1,36 +1,81 @@
 package ch.epfl.sweng.swenggolf;
 
+import android.support.test.espresso.intent.rule.IntentsTestRule;
 import android.support.test.rule.ActivityTestRule;
 import android.support.test.runner.AndroidJUnit4;
 
+import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
+import ch.epfl.sweng.swenggolf.database.Database;
+import ch.epfl.sweng.swenggolf.database.DatabaseUser;
+import ch.epfl.sweng.swenggolf.database.DbError;
+import ch.epfl.sweng.swenggolf.database.FakeDatabase;
+import ch.epfl.sweng.swenggolf.database.ValueListener;
 import ch.epfl.sweng.swenggolf.main.MainActivity;
+import ch.epfl.sweng.swenggolf.main.MainMenuActivity;
 
 import static android.support.test.espresso.Espresso.onView;
 import static android.support.test.espresso.action.ViewActions.click;
 import static android.support.test.espresso.action.ViewActions.closeSoftKeyboard;
 import static android.support.test.espresso.action.ViewActions.typeText;
 import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.intent.Intents.intended;
+import static android.support.test.espresso.intent.matcher.IntentMatchers.hasComponent;
+import static android.support.test.espresso.matcher.ViewMatchers.withContentDescription;
 import static android.support.test.espresso.matcher.ViewMatchers.withId;
 import static android.support.test.espresso.matcher.ViewMatchers.withText;
-
+import static org.junit.Assert.assertEquals;
 
 @RunWith(AndroidJUnit4.class)
 public class ProfileActivityTest {
+
+    User user = new User("Patrick", "Vetterli", "1234567890", "");
+
     @Rule
-    public final ActivityTestRule<MainActivity> mActivityRule =
-            new ActivityTestRule<>(MainActivity.class);
+    public final IntentsTestRule<MainActivity> mActivityRule =
+            new IntentsTestRule<>(MainActivity.class);
+
+    /**
+     * Initialise the Config and the Database for tests.
+     */
+    @Before
+    public void setUp(){
+        Config.isTest();
+        Config.setUser(new User(user));
+        Database database = new FakeDatabase(true);
+        Database.setDebugDatabase(database);
+    }
 
     @Test
     public void canEditUserName() {
         String newName = "Anonymous";
         onView(withId(R.id.profileButton)).perform(click());
+        onView(withId(R.id.name)).check(matches(withText(user.getUserName())));
```

Comment:
That is a good idea to check if the name matches, but it is perhaps better to do it in a separate test.